### PR TITLE
Updated Sampling documentation for Go tracing library

### PR DESCRIPTION
**Pull request by dianashevchenko**
### What does this PR do? What is the motivation?

This PR updates documentation around `DD_TRACE_SAMPLING_RULES` and `DD_SPAN_SAMPLING_RULES` in corresponding section of documentation on the Go library. The change is motivates by an upcoming release in dd-trace-go, that expands the support for the sampling rules. 
 

### Merge instructions

This PR should wait until the v1.60.0 release of dd-trace-go is out. Do NOT merge until then. You can check if the release is out here - https://github.com/DataDog/dd-trace-go/releases

Draft release notes - https://github.com/DataDog/dd-trace-go/releases/tag/untagged-42845413d23b769b4e97

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/tracing/trace_collection/library_config/go.md [hunk 1]
```diff
@@ -73,21 +73,24 @@ Overrides the default trace Agent port for DogStatsD metric submission. If the [
 
 `DD_TRACE_SAMPLING_RULES`
 : **Default**: `nil`<br>
-A JSON array of objects. Each object must have a `"sample_rate"`. The `"name"` and `"service"` fields are optional. The `"sample_rate"` value must be between `0.0` and `1.0` (inclusive). Rules are applied in configured order to determine the trace's sample rate.
+A JSON array of objects. Each object must have a `"sample_rate"`. The `"name"`,`"service"`, `"resource"` and `"tags"` fields are optional. The `"sample_rate"` value must be between `0.0` and `1.0` (inclusive). Rules are applied in configured order to determine the trace's sample rate.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_config/go.md:76]:
```suggestion
A JSON array of objects. Each object must have a `"sample_rate"`. The `"name"`,`"service"`, `"resource"`, and `"tags"` fields are optional. The `"sample_rate"` value must be between `0.0` and `1.0` (inclusive). Rules are applied in configured order to determine the trace's sample rate.
```

File: content/en/tracing/trace_collection/library_config/go.md [hunk 2]
```diff
@@ -73,21 +73,24 @@ Overrides the default trace Agent port for DogStatsD metric submission. If the [
...
+  - Set the sample rate to 40% for services that have `HTTP GET` resource name: `'[{"resource": "HTTP GET", "sample_rate": 0.4}]'`.
+  - Set the sample rate to 100% for services that have a `tier` tag with the value `premium`: `'[{"tags": {"tier":"premium"}, "sample_rate": 1}]'`.
 
 `DD_TRACE_SAMPLE_RATE`
 : Enable ingestion rate control.
 
 `DD_SPAN_SAMPLING_RULES`
 : **Default**: `nil`<br>
-A JSON array of objects. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
+A JSON array of objects. Each object must have a `"sample_rate"`. The `"name"`,`"service"`, `"resource"` and `"tags"` fields are optional. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
```
Comment (cswatt) [inline content/en/tracing/trace_collection/library_config/go.md:89]:
```suggestion
A JSON array of objects. Each object must have a `"sample_rate"`. The `"name"`,`"service"`, `"resource"`, and `"tags"` fields are optional. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
```

File: content/en/tracing/trace_pipeline/ingestion_mechanisms.md [hunk 3]
```diff
@@ -144,11 +144,20 @@ For example, to send 50% of the traces for the service named `my-service` and 10
 @env DD_TRACE_SAMPLING_RULES=[{"service": `my-service`, "sample_rate": 0.5}]
 ```
 
+Starting from version [v1.60.0][2], for Go applications, set by-resource and by-tags sampling rules with the `DD_TRACE_SAMPLING_RULES` environment variable.
+For example, to send 50% of the traces for the service named `my-service` and 1% of the traces for the resource `GET api/list_documents` and 10% of the rest of the traces:
```
Comment (cswatt) [inline content/en/tracing/trace_pipeline/ingestion_mechanisms.md:148]:
```suggestion
For example, to send 1% of the traces for the resource `GET api/list_documents` and 10% of the rest of the traces:
```

**Review by cswatt — DISMISSED** [review]

**Comment by dianashevchenko** [comment]
@cswatt hi, I'm not sure how to fix the alert breaking the page layout. Do you mind taking a look?
image

**Comment by cswatt** [comment]
I'll be looking into how best to fix this!

**Review by cswatt — APPROVED** [review]

**Pull request merged**